### PR TITLE
Pcg improvements

### PR DESCRIPTION
**Pull request by fcollman**
splits reads and writes to different services, and adds important environment variables to [support this PR](https://github.com/seung-lab/PyChunkedGraph/pull/457#issuecomment-1709350318)

### CODE REVIEW

**Comment by ranlu** [comment]
Thanks @fcollman. To match `t2d-standard-4` instances, I think it is more optimal setting `ZSTD_THREADS` to 4. Because 1 vcpu matches to 1 core for t2d instances. For other instance types like e2 we should use 2.

File: kubetemplates/ingress.yml [hunk 1]
```diff
@@ -21,20 +21,27 @@ metadata:
...
-    
+    nginx.ingress.kubernetes.io/use-regex: "true"
 #    ingress.kubernetes.io/whitelist-source-range: '128.112.0.0/16 140.180.0.0/16 63.237.233.8/32 73.97.166.124/30 24.56.235.125/32'
 spec:
   rules:
   - host: ${DOMAIN_NAME}
     http: &http_rules
       paths:
-      - path: /segmentation
+      - path: /segmentation/.*/(split|merge|split_preview)$
```
Comment (nkemnitz) [inline kubetemplates/ingress.yml:31]: Can't `split_preview` be part of the read-services? Conversely, `undo`, `redo` and `rollback` all involve write operations - do you need to add those here?

Comment (fcollman) [reply kubetemplates/ingress.yml:31]: yes good catch

**Comment by fcollman** [comment]
> Thanks @fcollman. To match `t2d-standard-4` instances, I think it is more optimal setting `ZSTD_THREADS` to 4. Because 1 vcpu matches to 1 core for t2d instances. For other instance types like e2 we should use 2.

We actually are deploying the same docker image on many different nodes for different services.. we have the write operations of segmentation, the read operations of segmentation, the mesh endpoints, and the mesh worker all running on different node types.  Should i modify this to match the number of CPUs available on each node type?

File: kubetemplates/ingress.yml (hunk 1 above)
Comment (fcollman) [reply kubetemplates/ingress.yml:31]: I think although you are right about split_preview it has memory requirements that are as high as split and so we want it on the beefier nodes.

Comment (fcollman) [reply kubetemplates/ingress.yml]: this is addressed

Comment (nkemnitz) [reply kubetemplates/ingress.yml]: Oh, I thought the difference is just the preemption mode, but all calls would be handled by faster n2d/t2d instances? You want to keep the read-only calls on the e2 instances?

**Comment by ranlu** [comment]
> We actually are deploying the same docker image on many different nodes for different services.. we have the write operations of segmentation, the read operations of segmentation, the mesh endpoints, and the mesh worker all running on different node types. Should i modify this to match the number of CPUs available on each node type?

The easy solution is to use a negative number, meaning using the same number of threads as logical CPUs on the machine. That is the original proposal, but @sdorkenw want to have better control over the multithreading behavior, so we added the env to specify the number of threads explicitly. We can start from a common `-1` and finetune the number of threads if it mess up any tasks. Of course you can assign different value to different worker type if you prefer that way.

**Pull request merged**